kuiper-game/Domain/Ship/MiningScan.cs

```diff
@@ -1,33 +1,43 @@
 using System;
-using System.Linq;
-using Kuiper.Domain;
-using Newtonsoft.Json;
-
-class MiningScan : IShipEvent
+using System.Collections.Generic;
+using Kuiper.Services;
+namespace Kuiper.Domain
 {
-    public MiningScan(Ship ship, DateTime startTime)
-    { 
-        StartTime = startTime;
-        Ship = ship;
-        TaskDuration = TimeSpan.FromHours(7);
-    }
-
-    public string Name => "Course set";
-
-    public string Description => $"Ship has started journey towards {Destination.Name}";
-
-    [JsonIgnore]
-    public Ship Ship {get;}
-    public Location Source { get; }
-    public Location Destination { get; }
-
-    public DateTime StartTime {get;}
-
-    public TimeSpan TaskDuration {get;}
-
-    public string GetPrompt()
+    public class MiningScan : IShipEvent
     {
-        return $"{Ship.Name} will finish scanning at {GameTime.Now().Add(TaskDuration)} ";
-
+        private const int chance = 80;
+        public MiningScan(DateTime startTime)
+        { 
+            StartTime = startTime;
+            TaskDuration = TimeSpan.FromHours(7);
+        }
+
+        public string Name => "Asteroid Scan";
+
+        public string Description => $"Scan the local planetary area for valuable asteroids";
+
+        public DateTime StartTime {get;}
+
+        public TimeSpan TaskDuration {get;}
+
+        public string StartEvent()
+        {
+            return $"{SolarSystemLocator.SolarSystem.Captain.Ship.Name} will finish scanning at {GameTime.Now().Add(TimeSpan.FromHours(7))} ";
+        }
+
+        public string EndEvent()
+        {
+            if(DiceFactory.Roller().D100(chance,0))
+            {
+                var distance = DiceFactory.Random().Next(50000, 1500000);
+                var asteroid = CelestialBodies.Asteroid.Create("Scanned Asteroid",3.2, CelestialBodies. ) 
```

Comment:
@Nicoolai i am trying to create a new asteroid, but i don't know if distance is in AU or km? 

How do i access the already existing bodies? Could there be a convinience static missing like Bodies.Sun or something?

Would it make more sense to be able to call a method in the SolarSystemService to create the Asteroid?

Does the asteroid need to be persisted down into the JSON with the rest of hte bodies?

Reply:
Yea, there's a lot of things I decided not to implement because I thought it was getting too big and started to have a lot of feature creep. 
Things that should probably be their own PRs afterwards.

Distance is in AU.
If we use KM, I just thought the numbers becomes really high, really fast.
But the GetDistance method has an inAU and an inKM.
Maybe we should also have a DistanceConverterService or something instead of the consts I've dropped in a couple of files :) (such a slacker)

So, the one method I did implement, was getting bodies by name, from `SolarSystemService`.
So you can call `SolarSystemService.GetBody("Sun")` to get the sun body. 
But it definitely might make sense to have an AddBody in there as well.

The JSON structure I imagined was just the game data. So basically what it uses to initially create the solarsystem. All the things we want the solarsystem to have, when a game starts.
I even made it so you could create other solarsystems, and save them as `.solarsystem` json files. Maybe not necessary but why not.

Reply:
That seems to have did the trick. I extended GetBodies to a method that returned all of them. Running tests now